.travis/test-coverage.sh

```diff
@@ -13,23 +14,25 @@ testCover() {
 	# switch to the directory to check
 	pushd $d > /dev/null
 	# create the coverage profile
-	coverageresult=`go test -v -coverprofile=$PROFILE_OUT`
+	coverageresult=$(go test $TAGS -coverprofile="${PROFILE_OUT}" -covermode=${MODE})
 	# output the result so we can check the shell output
 	echo ${coverageresult}
 	# append the results to acc.out if coverage didn't fail, else set the retval to 1 (failed)
-	( [[ ${coverageresult} == *FAIL* ]] && retval=1 ) || ( [ -f $PROFILE_OUT ] && grep -v "mode: set" $PROFILE_OUT >> $ACC_OUT )
+	( [[ ${coverageresult} == *FAIL* ]] && retval=1 ) || ( [ -f "${PROFILE_OUT}" ] && grep -v "mode: ${MODE}" "${PROFILE_OUT}" >> "${ACC_OUT}" )
 	# return to our working dir
 	popd > /dev/null
 	# return our return value
 	return $retval
 }
 
-# Init acc.out
-echo "mode: set" > $ACC_OUT
+# Init coverage.txt
+echo "mode: ${MODE}" > $ACC_OUT
 
-# Run test coverage on all directories containing go files
-find . -maxdepth 10 -type d | while read d; do testCover $d || exit; done
+# Run test coverage on all directories containing go files.
+find . -type d | while read d; do testCover $d || exit; done
 
 # Upload the coverage profile to coveralls.io
-[ -n "$COVERALLS_TOKEN" ] && ( goveralls -coverprofile=$ACC_OUT -service=travis-ci -repotoken $COVERALLS_TOKEN || echo -e '\n\e[31mCoveralls failed.\n' )
+[ -n "$COVERALLS_TOKEN" ] && ( goveralls -coverprofile=$ACC_OUT || echo -e '\n\e[31mCoveralls failed.\n' )
```

Comment:
What is the rule for enclosing the variables in `{ }`? Sometimes it's done, sometimes not.

Reply:
I'm doing it more than I really need to here. In string appends, the braces provide a lexical break. I tend to use them in quoted strings generally though.